jax/lax/lax.py

```diff
@@ -4803,7 +4811,23 @@ def _prescan_power_of_two(x, axis: int, op: Callable, unit):
   return x, total
 
 
-def _parallel_prefix_scan(x, axis: int, op: Callable, unit):
+def _parallel_prefix_scan(x, axis: int, op: Callable):
+  if op is max:
+    if onp.issubdtype(x.dtype, onp.integer):
+      unit = onp.iinfo(x.dtype).min
+    else:
+      unit = -onp.inf
```

Comment:
We should cast `np.inf` to the correct dtype here.

Also, does `bool` need to be handled?

Reply:
0 and 1 are also (and were not) cast to the correct dtype here, it happens later.  I'm not sure I see why it needs to be done explicitly here.

Reply:
as for bool, currently these ops (or cumsum / cumprod) aren't tested on booleans; I would argue that means it doesn't need to be handled.

Reply:
If the type doesn't matter, could we simplify this logic by passing -inf and +inf regardless of the data type?

Reply:
Casting inf to integer types seems like a bad idea (I believe it is technically undefined), and in this particular case both +inf and -inf map to the same integer value.

>>> a
array([inf])
>>> a.astype(np.int32)
array([-2147483648], dtype=int32)

>>> a
array([-inf])
>>> a.astype(np.int32)
array([-2147483648], dtype=int32)

Reply:
`cumsum` and `cumprod` work as expected on bool inputs (casting to int); I suspect we have users who rely on this, particularly for `cumsum`, so it would be worth adding those tests.

Reply:
I'm happy to do that in another PR if you think it's out of scope here.

Reply:
Ah, I see, the casting happens implicitly in `_prescan_power_of_two`.

I still think making sure this does something sane for boolean input is important.

Reply:
I added support for setting unit if the dtype is boolean, I'll leave adding the appropriate test to you, it does seem out of scope here.